site/content/how-to/traffic-management/snippets.md

```diff
@@ -0,0 +1,462 @@
+
+---
+
+## Setup
+
+- To enable Snippets, [install]({{< relref "/installation/" >}}) NGINX Gateway Fabric with these modifications:
+  - Using Helm: set the `nginxGateway.snippetsFilters.enable=true` Helm value.
+  - Using Kubernetes manifests: set the `--snippets-filters` flag in the nginx-gateway container argument, add `snippetsfilters` to the RBAC
+    rules with verbs `list` and `watch`, and add `snippetsfilters/status` to the RBAC rules with verb `update`. See this [example manifest](https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/main/deploy/snippets-filters/deploy.yaml) for clarification.
+
+- Save the public IP address and port of NGINX Gateway Fabric into shell variables:
+
+   ```text
+  GW_IP=<ip address>
+  GW_PORT=<port number>
+  ```
+
+  {{< note >}} In a production environment, you should have a DNS record for the external IP address that is exposed, and it should refer to the hostname that the gateway will forward for. {{< /note >}}
+
+- Create the coffee and tea example applications:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/app.yaml
+  ```
+
+- Create a Gateway:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/gateway.yaml
+   ```
+
+- Create HTTPRoutes for the coffee and tea applications:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/httproutes.yaml
+   ```
+
+- Test the configuration:
+
+  You can send traffic to the coffee and tea applications using the external IP address and port for NGINX Gateway Fabric.
+
+  Send a request to coffee:
+
+  ```shell
+  curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee
+  ```
+
+  This request should receive a response from the coffee Pod:
+
+  ```text
+  Server address: 10.244.0.7:8080
+  Server name: coffee-76c7c85bbd-cf8nz
+  ```
+
+  Send a request to tea:
+
+  ```shell
+  curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/tea
+   ```
+
+  This request should receive a response from the tea Pod:
+
+  ```text
+  Server address: 10.244.0.6:8080
+  Server name: tea-76c7c85bbd-cf8nz
+  ```
+
+  Before we enable rate limiting, try sending multiple requests to coffee:
+
+  ```shell
+  for i in `seq 1 10`; do curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee; done
+  ```
+
+  You should see all successful responses in quick succession as we configured any rate limiting rules yet.
+
+---
+
+## Configure rate limiting to the coffee application
+
+Configure rate limiting to the coffee application by adding the following `SnippetsFilter`:
```

Comment:
I'm concerned the wording and organization of this section implies a 1-1 relationship between a SnippetsFilter and a route. 

The intention of SnippetsFilters is that they are common configuration provided by the cluster operator or maybe even a devops team.  This person or team creates SnippetsFilters for stuff like rate-limiting or access control, that can be used by any application team. The app dev just has to reference the SnippetsFilter in their route. 

Can we model that intention in this doc through a bit of reorganization?

I'm thinking:

1. Create rate limiting SnippetsFilters -- one with nodelay and one without
2. Make sure SnippetsFilters are valid
3. App developer for coffee applies the nodelay SnippetsFilter and makes sure it works
4. App developer for tea spplies the regular SnippetsFilter and makes sure it works

Reply:
Added reorganization changes, also changed so the regular rate limiting SF occurs first, then we showcase the nodelay setting. Felt more logical like that to me